pages/apps/index.tsx

```diff
@@ -37,41 +34,25 @@ export default function Apps({
             </Link>
           ))}
         </div>
-        <ApplicationSections
-          popular={popular}
-          recentlyUpdated={recentlyUpdated}
-          editorsChoiceApps={editorsChoiceApps}
-          editorsChoiceGames={editorsChoiceGames}
-        ></ApplicationSections>
+        {topAppsByCategory.map((sectionData, i) => (
+          <ApplicationSection key={`categorySection${i}`} href={`/apps/category/${encodeURIComponent(sectionData.category)}`} applications={sectionData.apps} title={categoryToName(sectionData.category)}></ApplicationSection>
+        ))}
       </div>
-    </Main>
+    </Main >
   )
 }
 
 export const getStaticProps: GetStaticProps = async () => {
-  const recentlyUpdated = await fetchCollection(
-    Collections.recentlyUpdated,
-    APPS_IN_PREVIEW_COUNT
-  )
-  const editorsChoiceApps = await fetchCollection(
-    Collections.editorsApps,
-    APPS_IN_PREVIEW_COUNT
-  )
-  const editorsChoiceGames = await fetchCollection(
-    Collections.editorsGames,
-    APPS_IN_PREVIEW_COUNT
-  )
-  const popular = await fetchCollection(
-    Collections.popular,
-    APPS_IN_PREVIEW_COUNT
-  )
+  let topAppsByCategory: { category: string, apps: Appstream[] }[] = [];
+
+  const categoryPromise = Object.keys(Category).map(async (category: Category) => {
+    return { category, apps: await (await fetchCategory(category)).slice(0, APPS_IN_PREVIEW_COUNT) }
```

Comment:
the api doesn't have a limit option?

Reply:
Not yet, I've tried to get away without api changes. Mostly cause I can :)

And it should only matter at build time in this case.

Reply:
Amended this to depend on https://github.com/flathub/backend/pull/111